records_mover/records/pandas/prep_for_csv.py

```diff
@@ -21,13 +21,20 @@ def _convert_series_or_index(series_or_index: T,
                              field: RecordsSchemaField,
                              records_format: DelimitedRecordsFormat,
                              processing_instructions: ProcessingInstructions) -> Optional[T]:
+    if isinstance(series_or_index, pd.Series):
+        try:
+            first_item = series_or_index.iloc[0]
+        except IndexError:
+            first_item = series_or_index.dtype.type()
```

Comment:
can we not just do this in every case?

Reply:
Strangely, no. The reason this like 'picking first item' `series_or_index[0]` happens in the first place is that either the data type isn't specified for the series or index _or_ the first item might have a different type from what's specified in the series or index.

Reply:
That's why you have these weird loggers:
```python
            logger.debug("Dtype is %s, first element type %s", series_or_index.dtype,
                         type(first_item))

            logger.warning(f"Found {series_or_index.name} as unexpected "
                           f"type {type(first_item)}")
```